### DISCUSSION TITLE
compiler problem FundMe.sol

### DISCUSSION THREAD

**Discussion by fricpto** [smartcontractkit/full-blockchain-solidity-course-py#1812] [Q&A] [answered]
I copied and pasted FundMe.sol from the repository and it says that it needs different compiler version 
<img width="702" alt="image" src="https://user-images.githubusercontent.com/100000316/208554739-bd995b52-e794-45c0-b46f-bfb7c7a62d5b.png">
But in remix it works.

**Comment by propeltrain** [comment]
Try to manually pick the *compiler* to 0.6.6+….Then press  the “Compile FundMe.sol” tab

![D197E335-B490-4238-A079-C635A508CE2E](https://user-images.githubusercontent.com/109322953/208557672-0873445b-c3f5-4359-b599-b16d12fd060a.png)

**Comment by cromewar** [comment]
Hello, this might happen due to the extension configuration on vscode, try with this options.

1. Delete the build folder on for project.
2. Change the compiler version for solidity on the vscode extension, just right click and change the version.
3. go to your home folder, in windows `C:\Users\yourUser` there is a hidden folder called `.brownie`, delete it a try again.
4. If nothing works , try changing the vscode extension for solidity for the one from Hardhat.

**Comment by mattjaf** [comment]
is that the error in your contract where it has the red underline and and says that error in your screen shot in your console?

try picking a different compiler version your extension.

![image](https://user-images.githubusercontent.com/5776918/208833267-b29329e3-0593-43ad-8618-7097b0b672cb.png)

**Comment by fricpto** [comment, marked as answer]
Hello @cromewar  I have tried your suggestion and I've changed the compiler version many times but unfortunately nothing worked now I have these errors:
<img width="641" alt="Screenshot 2022-12-26 191833" src="https://user-images.githubusercontent.com/100000316/209571281-2120d575-64fb-41e1-9d85-d7170a2a6b41.png">
P.S: I have also restored the `.brownie` and the build folder and I don't have a hardhat vscode extension I only have a solidity extension

**Reply by cromewar** [reply]
You can replace the solidity extension and replace it with the hard hat one. Also can you share your brownie config please, sometimes the problem is with the version you are referring to on the config file.

**Reply by fricpto** [reply]
@cromewar I installed the hardhat one but my brownie config file is in `brownie simple storage`  folder

**Reply by fricpto** [reply]
@cromewar I watched a bit more of the course and now I added a `brownie config file` and my errors are still the same.
<img width="485" alt="Screenshot 2022-12-27 135743" src="https://user-images.githubusercontent.com/100000316/209663419-643c2107-3809-4035-a53c-80d9e8127957.png">
and this is what happens when I compile:
<img width="625" alt="Screenshot 2022-12-27 135918" src="https://user-images.githubusercontent.com/100000316/209663676-3ec9ac1d-eeec-4743-a85a-d2406fa8fe17.png">

**Reply by fricpto** [reply]
@cromewar  should I change my compiler version my current one is 0.6.6

**Reply by cromewar** [reply]
Remember all the projects should have their own `brownie-config` file, and yes you should change the compiler version on the extension, just right click when you have the solidity contract opened and change it to he version you need.

**Reply by fricpto** [reply]
@cromewar I switched to the normal solidity vscode extension and with the `brownie-config.yaml` added the problems disappeared. Thanks for your help.

**Comment by fricpto** [comment]
@mattjaf @cromewar @propeltrain when I uncheck this box the problems go do you suggest me do always uncheck this box:
<img width="553" alt="Screenshot 2022-12-26 223805" src="https://user-images.githubusercontent.com/100000316/209582182-ace7b9f6-0f78-4b6d-a5e8-f504117c8d10.png">

**Reply by mattjaf** [reply]
Nah don't disable that, switch your compiler to the appropriate version or use the carrot ^ in your contract when you specify your compiler version in the contract or simply ignore the error. 


The extension give you feedback as you type which can be helpful.

**Reply by fricpto** [reply]
@mattjaf if I ignored the problem I won't face any problem in the brownie fund me lesson?

**Reply by mattjaf** [reply]
is it currently breaking anything?

BTW what is your `Remote compiler` set to in your extension settings

**Reply by cromewar** [reply]
This is also a good work around @mattjaf, thanks.

**Reply by fricpto** [reply]
@mattjaf  it is 0.6.6

**Reply by fricpto** [reply]
@mattjaf  the problem was that I didn't add a `brownie-config.yaml` file. Thanks for your help.
